src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private Node<T> tail;
+    private int size;
 
+    public void push(T value) {
+        tail = new Node<>(tail, value);
+        size++;
     }
 
     public T peek() {
-        return null;
+        return stackIsNotEmpty() ? tail.value : null;
```

Comment:
```suggestion
        checkStackSize();
        return tail.value;
```

the problem is that you will never return null - this is a bit confusing